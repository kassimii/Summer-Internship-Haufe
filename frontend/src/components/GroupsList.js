import React from "react";

import GroupCard from "./GroupCard";

const GroupsList = (props) => {
  return props.groups ? (
    <div className="album py-5 bg-light">
      <div className="container">
        <div className="row">
          {props.groups.map((group) => {
            return (
              <div key={group.id} className="col-md-4">
                <div className="card mb-4 shadow-sm">
                  <GroupCard group={group} />
                </div>
              </div>
            );
          })}
        </div>
      </div>
    </div>
  ) : (
    <div>Loading....</div>
  );
};

export default GroupsList;
