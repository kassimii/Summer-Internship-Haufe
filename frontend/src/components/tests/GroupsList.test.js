import React from "react";
import { shallow, configure } from "enzyme";
import renderer from "react-test-renderer";
import GroupsList from "../GroupsList";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { Router } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("GroupsList", () => {
  const testGroups = [
    {
      id: "1",
      name: "Group1",
      claimToGroupMapping: ["claim1", "claim2"],
      defaultSettings: [
        {
          key: "setting1",
          value: "true"
        },
        {
          key: "setting2",
          value: "false"
        }
      ]
    },
    {
      id: "2",
      name: "Group2",
      claimToGroupMapping: ["claim1"],
      defaultSettings: [
        {
          key: "setting1",
          value: "false"
        },
        {
          key: "setting2",
          value: "false"
        },
        {
          key: "flag1",
          value: "true"
        }
      ]
    },
    {
      id: "3",
      name: "Group3",
      claimToGroupMapping: ["claim1", "claim2", "claim3"],
      defaultSettings: [
        {
          key: "setting3",
          value: "true"
        },
        {
          key: "setting2",
          value: "false"
        },
        {
          key: "setting4",
          value: "false"
        }
      ]
    },
    {
      id: "4",
      name: "Group4",
      claimToGroupMapping: ["claim1", "claim2", "claim5"],
      defaultSettings: [
        {
          key: "setting1",
          value: "true"
        }
      ]
    }
  ];

  it("renders with no group", () => {
    let groupsList = renderer.create(<GroupsList />);
    let tree = groupsList.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with with many groups", () => {
    const groupsList = shallow(<GroupsList groups={testGroups} />);
    expect(groupsList.find("div").length).toEqual(3 + 2 * testGroups.length);
  });
});
