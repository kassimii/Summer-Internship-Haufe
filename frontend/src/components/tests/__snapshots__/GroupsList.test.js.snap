// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsList renders with no group 1`] = `
<div>
  Loading....
</div>
`;

exports[`GroupsList renders with no group 2`] = `
<div>
  Loading....
</div>
`;

exports[`GroupsList renders with with many groups 1`] = `
<div>
  Loading....
</div>
`;
